saucal/woocommerce-contribution
# Fix/27030

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



Closes #27030 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create 2 Shipping Classes and 2 Shipping Zones (with Flat Rate added to each, we could name them New York and Florida).
2. Use the "Per Class" calculation type and only set rates for 1 class in each zone.
3. Create 2 products one adding the New York shipping class and the Florida to the other one
4. Add them to the cart individually, you should be able to complete your checkout if your shipping address matches the product's shipping class.
5. Add them to the cart together, you should never be able to complete your checkout. You should see that there are no shipping options available.
